src/util/metadata/vo_bit/mod.rs

```diff
@@ -222,3 +223,23 @@ pub fn is_internal_ptr_from_vo_bit<VM: VMBinding>(
 pub unsafe fn is_vo_addr(addr: Address) -> bool {
     VO_BIT_SIDE_METADATA_SPEC.load::<u8>(addr) != 0
 }
+
+#[cfg(feature = "vm_space")]
+use crate::MMTK;
+
+/// Set VO bit for a VM space object.
+/// Objects in the VM space are allocated/managed by the binding. This functio provides a way for
+/// the binding to set VO bit for an object in the space.
+#[cfg(feature = "vm_space")]
+pub fn set_vo_bit_for_vm_space_object<VM: VMBinding>(
+    mmtk: &'static MMTK<VM>,
+    object: ObjectReference,
+) {
+    use crate::policy::space::Space;
+    debug_assert!(
+        mmtk.get_plan().base().vm_space.in_space(object),
+        "{} is not in VM space",
+        object
+    );
+    set_vo_bit(object);
```

Comment:
This sets a single VO bit at a time.  Depending on the density of objects in the VM space, if this is a bottleneck, we can provide another function that allows the VM binding to provide a bitmap directly, and memcpy into our VO bit metadata (in a single pass, or chunk by chunk).

Reply:
I agree, but I feel this is out of scope for this PR.